airmozilla/popcorn/views.py

```diff
@@ -51,11 +42,68 @@ def event_meta_data(request):
 
     assert response.status_code == 302, response.status_code
 
+    purl = urlparse(response.headers['location'])
+    location = r'{0}://{1}{2}'.format(purl.scheme, purl.hostname, purl.path)
+
+    return location
+
+
+def get_thumbnail(event):
+    image = event.picture and event.picture.file or event.placeholder_img
+    geometry = '160x90'
+    crop = 'center'
+
+    return thumbnail(image, geometry, crop=crop)
+
+
+@add_cors_header('*')
+@json_view
+def event_meta_data(request):
+    slug = request.GET.get('slug')
+    event = get_object_or_404(Event, slug=slug)
```

Comment:
You're using a `.get()`, which implies that you're OK with that it might become `slug=None`. Then you can't do the next line. The `get_object_or_404`. 

I would change these two lines to:
```python
if not request.GET.get('slug'):
    return http.HttpResponseBadRequest('slug')
event = get_object_or_404(Event, slug=request.GET['slug'])
```